lib/core/src/Cardano/Wallet/Transaction.hs

```diff
@@ -289,6 +289,10 @@ data TransactionLayer k tx = TransactionLayer
         --    - Any increase in cost is covered:
         --        If the total cost has increased by 𝛿c, then the fee value
         --        will have increased by at least 𝛿c.
+        --
+        -- If the provided surplus is greater or equal to
+        -- 'maximumCostOfIncreasingCoin', this function will never return
+        -- 'ErrMoreSurplusNeeded'.
```

Comment:
I'm not sure about this wording. It tells the caller how they can guarantee to not get an error, but it implies that there is a grey area where if they provide a slightly smaller surplus, then they might get an error, or they might not.

I wonder if we can turn this into a more concrete statement?

For example:

```suggestion
        -- If the provided surplus is <insert relevant condition>
        -- this function will return 'ErrMoreSurplusNeeded'.
```

Reply:
> It tells the caller how they can guarantee to not get an error

That's what is important for `balanceTransaction`, and what I wanted to convey.

I believe the exact behaviour of the "grey area" is an implementation detail. 

Still, we could add something generic:

```
If cost of distributing the provided surplus is greater than the surplus itself,
the function will return 'ErrMoreSurplusNeeded'. If the provided surplus satisfies
`>= maximumCostOfIncreasingCoin feePolicy`, the function will always return 'Right'.
```

Thoughts?